src/services/types.ts

```diff
@@ -19,34 +19,34 @@ namespace ts {
         getFirstToken(sourceFile?: SourceFile): Node;
         getLastToken(sourceFile?: SourceFile): Node;
         // See ts.forEachChild for documentation.
-        forEachChild<T>(cbNode: (node: Node) => T, cbNodeArray?: (nodes: Node[]) => T): T;
+        forEachChild<T>(cbNode: (node: Node) => T, cbNodeArray?: (nodes: Node[]) => T): T | undefined;

```

Comment:
@andy-ms 
`T` is inferred from the return type of the callback(s). I don't get why it should be declared to return `T | undefined`. Can you please explain your thoughts on this?